ankerctl.py

```diff
@@ -434,5 +435,86 @@ def config_show(env):
             print(f"    p2p_hosts: {', '.join(p.p2p_hosts)}")
 
 
+@main.group("webserver", help="Built-in webserver support")
+@pass_env
+def webserver(env):
+    env.require_config()
+
+
+app = Flask(__name__)
+
+
+@app.get("/")
+def app_root():
+    prefix = "https://github.com/Ankermgmt"
+    return f"""<html>
+    <head><title>ankerctl web server</title></head>
+    <body>
+    <a href=\"{prefix}/ankermake-m5-protocol\">{prefix}/ankermake-m5-protocol</a>
+    <h1>Connecting prusaslicer:</h1>
+    <ul>
+    <li>Select "window" menu, "Printer Settings Tab"</li>
+    <li>Select gear icon ("Edit Physical Printer")</li>
+    <li>Input these settings</li>
+    <ul>
+    <li>Host type: OctoPrint</li>
+    <li>Hostname, IP or URL: localhost:{app.config["port"]}</li>
+    </ul>
+    <li>Select "OK"</li>
+    <li>Enjoy printing using ankerctl :)</li>
+    </ul>
+    </html>"""
+
+
+@app.get("/api/version")
+def app_api_version():
+    return {
+        "api": "0.1",
+        "server": "1.3.10",
+        "text": "OctoPrint 1.3.10"
+    }
+
+
+@app.post("/api/files/local")
+def app_api_files_local():
+    env = app.config["env"]
+
+    user_name = request.headers.get("User-Agent", "ankerctl").split("/")[0]
+
+    no_act = not cli.util.parse_http_bool(request.form["print"])
+
+    if no_act:
+        cli.util.http_abort(409, "Upload-only not supported by Ankermake M5")
+
+    fd = request.files["file"]
+
+    api = cli.pppp.pppp_open(env)
+
+    data = fd.read()
+    fui = FileUploadInfo.from_data(data, fd.filename, user_name=user_name, user_id="-", machine_id="-")
+    log.info(f"Going to upload {fui.size} bytes as {fui.name!r}")
+    try:
+        cli.pppp.pppp_send_file(api, fui, data)
+        log.info("File upload complete. Requesting print start of job.")
+        api.aabb_request(b"", frametype=FileTransfer.END)
+    except PPPPError as E:
+        log.error(f"Could not send print job: {E}")
+    else:
+        log.info("Successfully sent print job")
+    finally:
+        api.stop()
+
+    return {}
+
+
+@webserver.command("run", help="Run ankerctl webserver")
+@pass_env
+def webserver(env, port=4470):
```

Comment:
Is 4470 a flask default port? Or is it just a random port? 

What if we used a port more resonant with the M5 naming 😄, such as `5555`

Reply:
Not completely random, but I'm open to suggestions :)

```python
sum([ord(c) - ord('a') for c in "anker"]) == 44
sum([ord(c) - ord('a') for c in "ankermake"]) == 70
```